test/error_messages.coffee

```diff
@@ -989,3 +995,160 @@ test "`&&=` and `||=` with a space in-between", ->
     a or = 1
          ^
   '''
+
+test "anonymous functions cannot be exported", ->
+  assertErrorFormat '''
+    export ->
+      console.log 'hello, world!'
+  ''', '''
+    [stdin]:1:8: error: unexpected ->
+    export ->
+           ^^
+  '''
+
+test "anonymous classes cannot be exported", ->
+  assertErrorFormat '''
+    export class
+      constructor: ->
+        console.log 'hello, world!'
+  ''', '''
+    SyntaxError: Unexpected token export
+  '''
+
+test "unless enclosed by curly braces, only * can be aliased", ->
+  assertErrorFormat '''
+    import foo as bar from 'lib'
+  ''', '''
+    [stdin]:1:12: error: unexpected as
+    import foo as bar from 'lib'
+               ^^
+  '''
+
+test "unwrapped imports must follow constrained syntax", ->
+  assertErrorFormat '''
+    import foo, bar from 'lib'
+  ''', '''
+    [stdin]:1:13: error: unexpected identifier
+    import foo, bar from 'lib'
+                ^^^
+  '''
+  assertErrorFormat '''
+    import foo, bar, baz from 'lib'
+  ''', '''
+    [stdin]:1:13: error: unexpected identifier
+    import foo, bar, baz from 'lib'
+                ^^^
+  '''
+  assertErrorFormat '''
+    import foo, bar as baz from 'lib'
+  ''', '''
+    [stdin]:1:13: error: unexpected identifier
+    import foo, bar as baz from 'lib'
+                ^^^
+  '''
+
+test "cannot export * without a module to export from", ->
+  assertErrorFormat '''
+    export *
+  ''', '''
+    [stdin]:1:9: error: unexpected end of input
+    export *
+            ^
+  '''
+
+test "imports and exports must be top-level", ->
+  assertErrorFormat '''
+    if foo
+      import { bar } from 'lib'
+  ''', '''
+    [stdin]:2:3: error: import statements must be at top-level scope
+      import { bar } from 'lib'
+      ^^^^^^^^^^^^^^^^^^^^^^^^^
+  '''
+  assertErrorFormat '''
+    foo = ->
+      export { bar }
+  ''', '''
+    [stdin]:2:3: error: export statements must be at top-level scope
+      export { bar }
+      ^^^^^^^^^^^^^^
+  '''
+
+test "cannot import the same member more than once", ->
+  assertErrorFormat '''
+    import { foo, foo } from 'lib'
+  ''', '''
+    [stdin]:1:15: error: already imported
+    import { foo, foo } from 'lib'
+                  ^^^
+  '''
+  assertErrorFormat '''
+    import { foo, bar, foo } from 'lib'
+  ''', '''
+    [stdin]:1:20: error: already imported
+    import { foo, bar, foo } from 'lib'
+                       ^^^
+  '''
+  assertErrorFormat '''
+    import { foo, bar as foo } from 'lib'
+  ''', '''
+    [stdin]:1:15: error: already imported
+    import { foo, bar as foo } from 'lib'
+                  ^^^^^^^^^^
```

Comment:
You mean just rename "already imported" to "duplicate declaration"? I think that would work, technically "duplicate declaration" could refer to both importing the same symbol twice or trying to declare and assign the same symbol twice.